tests/installation/test_installer.py

```diff
@@ -1831,3 +1832,47 @@ def test_installer_can_handle_old_lock_files(
 
     # colorama will be added
     assert 8 == installer.executor.installations_count
+
+
+@pytest.mark.parametrize(
+    "quiet,lines",
+    [
+        (True, []),
+        (
+            False,
+            [
+                "Updating dependencies",
+                "Resolving dependencies...",
+                "",
+                "Writing lock file",
+                "",
+                "Package operations: 1 install, 0 updates, 0 removals",
+                "",
+                "  • Installing a (1.0)",
+            ],
+        ),
+    ],
+)
+def test_quiet(package, pool, locker, env, installed, config, repo, quiet, lines):
+    io = BufferedIO()
+    io.set_quiet(quiet)
+
+    installer = Installer(
+        io,
+        env,
+        package,
+        locker,
+        pool,
+        config,
+        installed=installed,
+        executor=Executor(env, pool, config, io),
+    )
+    installer.use_executor(True)
+
+    package_a = get_package("A", "1.0")
+    repo.add_package(package_a)
+    package.add_dependency(Factory.create_dependency("A", "~1.0"))
+
+    installer.run()
+
+    assert io.fetch_output().splitlines() == lines
```

Comment:
```suggestion
    output = io.fetch_output().strip()
    assert (not output and quiet) or (output and not queit)
```

Testing output can be painful. We could do this instead since we do not really have to validate the content for these tests as they are verified elsewhere.